File: HackerRank/InterviewPreparationKit/c#/salesByMatch.DamianD.cs
static int sockMerchant(int n, int[] ar) {
    //Un dictionary es un hashmap
    Dictionary<int,int> map = new Dictionary<int,int>();
    int cont = 0;
    foreach(var num in ar)
    {
        //Si el dictionary no contiene la clave del numero, lo agrego
        if (!map.ContainsKey(num)) map.Add(num, 1);
        else map[num]++; //Caso contrario le sumo uno
        if (map[num] % 2 == 0) cont++; //Y si es par la cantidad de medias en cada num, sumo 1 al contador
    }
    return cont;

}

File: HackerRank/InterviewPreparationKit/js/Warm-up/Sales By Match/salesByMatch.DamianD.js
// Complete the sockMerchant function below.
// En este caso estoy creando un hashmap para agarrar cada valor y de esa forma poder indexar muy rapidamente
// La funcion map me permite acceder a cada elemento a la vez
function sockMerchant(n, ar) {
    let mapa = new Map();
    let cont = 0;
    ar.map(x => {
        mapa[x] == null ? mapa[x] = 1 : mapa[x]++; // Operador ternario, en caso de que la condicion de la izquierda sea null (No hay valor para ese index)
        //Entonces lo asigno y le doy 1 (existe 1 de ese valor), caso contrario si ya existia, le agrego uno más
        if(mapa[x] % 2 === 0) cont++;
        //Cada vez que sea par el numero de medias que encuentra, conteo uno para ese indice
        });
    return cont;
}
//